fortnite_api/enums.py

```diff
@@ -454,4 +582,22 @@ def _from_str(cls: Type[Self], string: str) -> Self:
         # The Epic Games API "Product" enums contains both lower case and capitalized values, so we need to handle both.
         # To easily handle this, we'll remove the "Product." prefix and convert it to lowercase.
         trimmed = string.split('.')[-1]
-        return cls(trimmed.lower())
+        return try_enum(cls, trimmed.lower())
+
+
+def create_unknown_value(cls: Type[E], val: Any) -> E:
+    value_cls = cls._enum_value_cls_  # type: ignore # This is narrowed below
+    name = f'UNKNOWN_{val}'
+    return value_cls(name=name, value=val)
+
+
+def try_enum(cls: Type[E], val: Any) -> E:
+    """A function that tries to turn the value into enum ``cls``.
+
+    If it fails it returns a proxy invalid value instead.
+    """
+
+    try:
+        return cls._enum_value_map_[val]  # type: ignore # All errors are caught below
+    except (KeyError, TypeError, AttributeError):
+        return create_unknown_value(cls, val)
```

Comment:
All the type ignores can be removed here by doing the following,
- Remove the type checking patch at the top of this file (`if TYPE_CHECKING: from enum import Enum`) such that the custom `Enum` class is defined at all times,
- Correct the `_enum_value_map_` and related `ClassVars` by moving them to the `Enum` class definition, and
- Add the missing `_enum_value_cls` ClassVar on the `Enum` class definition.

This points `TypeVar('E', bound='Enum')` to the right type and, in doing so, doesn't convince the user that some enum in the library is `enum.Enum` when it isn't.

Let me know if this logic doesn't pan out, but I tested it locally and the type errors went away. Currently the typing indicates that,
```python
class MyEnum(Enum):
    FOO = 1

print(MyEnum.__class__._enum_value_map_)
```
is valid syntax, when in actuality it is `MyEnum._enum_value_map_`.

Reply:
Couldn't figure out a solution that doesn't cause more type errors. Feel free to implement your solution right within this PR.

Reply:
Appended in cf883e6 👍